### ISSUE TITLE
New aim assist functionalities

### ISSUE THREAD

**Issue by Boondorl** [UZDoom/UZDoom#1900]
Currently the only aim assist that UZDoom offers is the default Doom autoaim. To help improve accessibility and controller usage, some new methods should be added into the engine to give control at a global level without needing to rely on mods. This includes:

* Aim friction. When looking near a valid target, the aim becomes "heavier" to make it less likely to overshoot or steer off course.
* Modern bullet magnetism. Doom already has this, but it can be improved significantly to allow a full dynamic range instead of just at limited angles horizontally and only for some weapons.
* Snap aiming. When pressing a button, snap to the nearest valid target next to the crosshair. Most modern games do this when aiming down sight.
* Aim magnetism. Keep the aim centered on the current target as best as possible without snapping. This is often paired with light automated tracking.
* Lock on. As an alternative to snap aiming or automated methods, allow holding the button to stay locked on to the target. A free aim mode while locked on would also be good to give the ability to lead shots without looking off target.

Some of these will be mutually exclusive with each other (e.g. snap aim vs lock on), but that can be figured out as things are added. All of these should be given options to have fuller control over how they're felt when in-game. Server options will also need to be present for multiplayer hosts so they can tune PvP appropriately.

**Comment by mc776** [comment]
This could be good for some fancy off-label camera stuff too...

**Comment by RexenSquared** [comment]
Some of the options that I think that would need to be added is make it only work for hostiles only, allow friendly fire, allow targeting props. The only issue is how to tell if something is an actual monster vs a misconfiged actor as I remember one time coming across an asset for mappers that had a set of props flag as being a monster. But I can still see this being useful for some people if done right.
